sdk/src/main/java/com/uid2/storage/FileStorageManager.kt

```diff
@@ -0,0 +1,60 @@
+package com.uid2.storage
+
+import android.content.Context
+import com.uid2.data.IdentityStatus
+import com.uid2.data.IdentityStatus.NO_IDENTITY
+import com.uid2.data.UID2Identity
+import java.io.File
+import java.io.InputStreamReader
+import java.io.OutputStreamWriter
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.json.JSONObject
+
+/**
+ * An implementation of the StorageManager that persists UID2Identity instances in clear-text via a File.
+ */
+internal class FileStorageManager(
+    private val identityFile: File,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+): StorageManager {
+
+    constructor(context: Context): this(File(context.filesDir, FILE_IDENTITY))
+
+    override suspend fun saveIdentity(identity: UID2Identity, status: IdentityStatus) = withContext(ioDispatcher) {
+        return@withContext runCatching {
+            OutputStreamWriter(identityFile.outputStream()).use { writer ->
+
+                // After converting the UID2Identity to JSON, we will extend it to also include the IdentityStatus.
+                val identityJson = identity.toJson()
+                identityJson.put(KEY_STATUS, status.value)
+
+                writer.write(identityJson.toString(0))
+                return@runCatching true
+            }
+        }.getOrDefault(false)
+    }
+
+    override suspend fun loadIdentity(): Pair<UID2Identity?, IdentityStatus> = withContext(ioDispatcher) {
+        return@withContext runCatching {
+            InputStreamReader(identityFile.inputStream()).use { reader ->
```

Comment:
Similarly, idiomatic way would be:

```
val jsonString = identityFile.readText(Charsets.UTF_8)
```

Reply:
Sure. If I've being explicit with the `Charset` during the read, I'll do that with the write too.